File: java assignments/Day02assignment/src/myownautoshop/Truck.java
package myownautoshop;

public class Truck extends Car{
	int weight ;
	public Truck(int a , double b , String c)
	{
		super(a,b,c);
	}
	public double getsalceprice()
	{
		double price ;
		if(weight>2000) {
			 price = regularprice - (regularprice*10/100);
		}
		else {
			 price = regularprice - (regularprice*20/100);
		}
		return price;
	}
	public void show()
	{
		System.out.println("Truck weight : " + weight);
		System.out.println("Truck speed : " + speed);
		System.out.println("Truck color : " + color);
		System.out.println("Truck SalePrice : " + getsalceprice());
		System.out.println();
	}

}

File: typescript examples/assignments/Truck.ts
import { Car } from "./Car";

class Truck extends Car{
    weight:Number;
    getSalePrice(): void {
        if(this.weight >2000){
            var prc = this.regularPrice - (this.regularPrice*.1);
            console.log(prc);
        }
        else{
            var prc = this.regularPrice - (this.regularPrice*.2);
            console.log(prc);
        }
    }
}

var t1 = new Truck();
t1.weight=3000;
t1.regularPrice=5000;
t1.speed = 100;
t1.color= "white";
t1.getSalePrice();